cmd/macpassd/dbConnection.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type MacRegistration struct {
+	mac  string
+	id   int
+	user string
+}
+
+func getActive(con *sql.DB) []MacRegistration {
+	selectOutdated := `select id,User,MAC from Log where ? > maxTime and isDown = true`
+	row, err := con.Query(selectOutdated, time.Now())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var macs []MacRegistration
+	var user string
+	var mac string
+	var id int
+	for row.Next() {
+		row.Scan(&id, &user, &mac)
+		macs = append(macs, MacRegistration{mac, 0, user})
+	}
+	return macs
+}
+func setOutdated(con *sql.DB) []MacRegistration {
+	selectOutdated := `select id,User,MAC from Log where ? < maxTime and isDown = false`
+	row, err := con.Query(selectOutdated, time.Now())
+	if err != nil {
+		log.Fatal(err)
+	}
+	var macs []MacRegistration
+	var user string
+	var mac string
+	var id int
+	for row.Next() {
+		row.Scan(&id, &user, &mac)
+		macs = append(macs, MacRegistration{mac, 0, user})
+	}
+	for i := range macs {
+		id := macs[i].id
+		updateOutdated := `update Log set isDown = true where id = ?`
+		con.Exec(updateOutdated, id)
+	}
+	return macs
+}
+
+func insertDB(user string, mac string, endTIme time.Time, duration time.Duration, db *sql.DB) {
+	var now = time.Now()
+	var maxTime = now.Add(duration)
+	const insertLog = `insert into Log(User, MAC, startTime, maxTime, isDown) values(?, ?, ?,? , false)`
+	db.Exec(insertLog, user, mac, now, maxTime)
+}
```

Comment:
Non mi è chiarissima la gestione delle variabili relative al tempo in questo metodo. In particolare io salvo, per ogni autenticazione, lo `start` e la `duration`. Lo `start` memorizza il momento in cui la connessione è stat attivata, mentre la `duration` indica quanto tempo deve rimanere connesso.

Fai in modo che la funzione prenda in input una struttura `macRegistration` definita dal file main e utilizza i sui campi. Probabilmente per farla vedere anche al file del db dovrei renderla globale mettendo maiuscola la prima lettera nel nome della struttura